datafusion/physical-expr/src/functions.rs

```diff
@@ -2872,4 +2874,17 @@ mod tests {
 
         Ok(())
     }
+
+    // Helper function
+    // The type coercion will be done in the logical phase, should do the type coercion for the test
```

Comment:
👍 

BTW this is basically what I need to do in IOx / why I made https://github.com/apache/arrow-datafusion/issues/3708